eval.c

```diff
@@ -236,7 +236,8 @@ sexp sexp_extend_synclo_env (sexp ctx, sexp env) {
       sexp_env_renames(e2) = sexp_env_renames(e1);
 #endif
     }
-    sexp_env_parent(e2) = sexp_context_env(ctx);
+    // ISSUE_296 - Dereference of null pointer, stored line 231
+    if (e2) { sexp_env_parent(e2) = sexp_context_env(ctx); }
```

Comment:
Can only be null if the alloc above failed, in which case we should return OOM.
It's been a standing TODO to go through and catch all possible OOM cases.